Syncgateway-replicate stopping on attachment with no content type

### Sync Gateway version
1.3.1

### Operating system
RedHat 7.2

### Config file
 "replications":[
   {
               "replication_id":"ptxdata_C1_C2",
               "source": "ptxdata",
               "target": "http://prod-SYNCGWREP-V1.service.2.com:4985/ptxdata/",
               "continuous":true,
               "changes_feed_limit":1000,
               "filter":"sync_gateway/bychannel",
               "query_params":["*"]
    }

### Error Log output
2017-03-13T18:20:33.469+05:30 Enabling logging: [Bucket CRUD CRUD+ HTTP HTTP+ Access Cache Shadow Shadow+ Changes Changes+ Events Events+]
_time=2017-03-13T18:20:33.495+05:30 _level=INFO _msg= Trying with selected node 1
_time=2017-03-13T18:20:33.495+05:30 _level=INFO _msg= Trying with http://192.168.193.136:8091/pools/default/bucketsStreaming/ptxdata
_time=2017-03-13T18:20:33.505+05:30 _level=INFO _msg=Got new configuration for bucket ptxdata
_time=2017-03-13T18:20:33.523+05:30 _level=INFO _msg= Trying with selected node 1
panic: runtime error: index out of range

goroutine 107 [running]:
github.com/couchbaselabs/sg-replicate.Replication.fetchBulkGet(0xc82019ae10, 0xd, 0xc82015de80, 0xc82019ae20, 0x7, 0xc82019b600, 0x1, 0x1, 0xc82015de00, 0xc82018acb6, ...)
    /home/couchbase/jenkins/workspace/sgw-unix-build/1.3.1/community/godeps/src/github.com/couchbaselabs/sg-replicate/synctube.go:472 +0x36a2
created by github.com/couchbaselabs/sg-replicate.stateFnActiveFetchRevDiffs
    /home/couchbase/jenkins/workspace/sgw-unix-build/1.3.1/community/godeps/src/github.com/couchbaselabs/sg-replicate/replication_state.go:159 +0x6f1

goroutine 1 [IO wait]:
net.runtime_pollWait(0x7fe5259a4680, 0x72, 0xc820098080)
    /usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8206e3b80, 0x72, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8206e3b80, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8206e3b20, 0x0, 0x7fe5259a5528, 0xc82028ede0)
    /usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820278598, 0x452c30, 0x0, 0x0)
    /usr/local/go/src/net/tcpsock_posix.go:254 +0x4d


### Expected behavior
sg-replicate should restart

### Actual behavior
Not starting with above log error

### Steps to reproduce
Issue is inconsistent. currently system is in same state. would be happy to provide if any info need.

--- a/manifest/default.xml
+++ b/manifest/default.xml
@@ -93,7 +93,7 @@
 
   <project name="crypto" path="godeps/src/golang.org/x/crypto" remote="couchbasedeps" revision="c89e5683853da5ed97731b507dcd8cda2b11afaf"/>
 
-  <project name="sg-replicate" path="godeps/src/github.com/couchbaselabs/sg-replicate" remote="couchbaselabs" revision="24b8252856d71a541cbfeb2203cf7c0ce3634b2b"/>
+  <project name="sg-replicate" path="godeps/src/github.com/couchbaselabs/sg-replicate" remote="couchbaselabs" revision="8c736c4ac8f42dadf0b1fa3dfe27dcba6537b900"/>
 
   <project name="clog" path="godeps/src/github.com/couchbase/clog" remote="couchbase" revision="eebc98233c3e032eb6b9036575d51324ab5932e6"/>
 

